Sources/SecureEnclave.swift

```diff
@@ -29,8 +29,8 @@ public final class SecureEnclave {
     public enum Result<Type: Equatable>: Equatable {
         /// Data was retrieved from the keychain.
         case success(Type)
-        /// User dismissed the user-presence prompt.
-        case userCancelled
+        /// User dismissed the user-presence prompt. If `dueToAuthorizationFailure` is true, the user either entered the wrong password or locked the device while the prompt was presented.
+        case userCancelled(dueToAuthorizationFailure: Bool)
```

Comment:
Is it worth breaking this out into a new case? Feels a bit odd to tack this on to the canceled state to me.

Reply:
It's worth reading up on https://github.com/square/Valet/pull/73 and https://github.com/square/Valet/issues/143. UserCancelled and AuthFailure have a good bit of overlap, and I don't think we should make our consumers tell the difference between them.

Specifically, AuthFailure can mean that the user locked their screen while the prompt was up, which feels a lot like a UserCancel even to me. We've been conflating UserCancelled and AuthFailure since we introduced the ability to detect UserCancel back in https://github.com/square/Valet/pull/73. What we're doing in this PR is giving our consumers the ability to tease these cases apart if they feel like it, without breaking existing adopters, and without adding burden to new adopters who don't care about the differences between these cases.

I think it's important to view the consumer-facing API without a whole lot of knowledge of how the underlying keychain works. There are currently 3 options in this enum, which I believe accurately define what happens when you request a value from the Secure Enclave:
1. `success` – i.e. "We got the value!"
2. `userCancelled` - i.e. "The user bailed out of the prompt somehow"
3. `itemNotFound` - i.e. "There is no item in the keychain"

In this world, authorization failure (i.e. "incorrect passcode" or "user locked the screen") feels like a subset of `userCancelled`. If I could tease "incorrect passcode" apart from "user locked the screen", I'd add another enum case. But since I can't, and I can't provide more insight than "we got an AuthFailed response", I'd rather keep auth failure as an associated value on the general concept of `userCancelled`. Does that make sense?

Reply:
👍

Reply:
So, the real question becomes: do we want to add this API? I feel a little weird exposing this kind of keychain-specific API; it violates some of our promise of "without knowing a thing about how the Keychain works". But, it could also be useful information per the request in #143. This API errs on the side of enabling our consumers to do what they need with our API, rather than keeping the API as simple as possible. But I think we may have struck a reasonable balance between simplicity for the average consumer and making complex use cases possible.

Does this API pass the smell test for you? Does it seem like we've struck a good balance / made a reasonable trade-off here?

Reply:
Added my higher level comment on #143